packages/openactive-integration-tests/test/features/approval/minimal-proposal/implemented/accept-proposal-book-test.js

```diff
@@ -31,101 +40,101 @@ FeatureHelper.describeFeature(module, {
-    it('should include an orderProposalVersion, of the form {orderId}/versions/{versionUuid}', () => {
-      expect(state.pResponse.body).to.have.property('orderProposalVersion')
-        .which.matches(RegExp(`${state.uuid}/versions/.+`));
-    });
-    // TODO does validator check that orderItemStatus is https://openactive.io/OrderItemProposed
-    // TODO does validator check that full Seller details are included in the seller response?
+  const p = new PFlowStage({
+    ...defaultFlowStageParams,
+    prerequisite: c2,
+    getInput: () => ({
+      orderItems: fetchOpportunities.getOutput().orderItems,
+      totalPaymentDue: c2.getOutput().totalPaymentDue,
+    }),
   });
-
-  describe('Simulate Seller Approval (Test Interface Action)', () => {
-    (new TestInterfaceAction({
-      flow,
-      logger,
+  const [simulateSellerApproval, orderFeedUpdate] = OrderFeedUpdateFlowStageUtils.wrap({
+    // FlowStage that is getting wrapped
+    wrappedStageFn: prerequisite => (new TestInterfaceActionFlowStage({
+      ...defaultFlowStageParams,
+      testName: 'Simulate Seller Approval (Test Interface Action)',
+      prerequisite,
       createActionFn: () => ({
         type: 'test:SellerAcceptOrderProposalSimulateAction',
         objectType: 'OrderProposal',
-        objectId: state.pResponse.body['@id'],
+        objectId: p.getOutput().orderId,
       }),
-      completedFlowStage: 'P',
-    }))
-      .beforeSetup()
-      .successChecks();
+    })),
+    // Params for the Order Feed Update stages
+    orderFeedUpdateParams: {
+      ...defaultFlowStageParams,
+      prerequisite: p,
+      testName: 'Order Feed Update (after Simulate Seller Approval)',
+    },
+  });
+  const b = new BFlowStage({
+    ...defaultFlowStageParams,
+    prerequisite: orderFeedUpdate,
+    getInput: () => ({
+      orderItems: fetchOpportunities.getOutput().orderItems,
+      totalPaymentDue: p.getOutput().totalPaymentDue,
+      orderProposalVersion: p.getOutput().orderProposalVersion,
+    }),
   });
 
-  describe('Orders Feed (after P)', () => {
-    const orderFeedUpdate = (new OrderFeedUpdate({
-      state,
-      flow,
-      logger,
-      ordersFeedMode: 'orders-feed-after-p',
-    }))
-      .beforeSetup()
-      .successChecks()
-      .validationTests();
+  // ## Set up tests
+  FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(fetchOpportunities);
 
-    it('should have orderProposalStatus: SellerAccepted', () => {
-      expect(orderFeedUpdate.getStateResponse().body).to.have.nested.property('data.orderProposalStatus', 'https://openactive.io/SellerAccepted');
-    });
-    it('should have orderProposalVersion same as that returned by P (i.e. an amendment hasn\'t occurred)', () => {
-      expect(orderFeedUpdate.getStateResponse().body).to.have.nested.property('data.orderProposalVersion', state.pResponse.body.orderProposalVersion);
-    });
+  FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(c1, {
+    itAdditionalTests() {
```

Comment:
Just realised this should be called `additionalTests` not `itAdditionalTests`, as `itAdditionalTests` implies it has a built-in `it()` (as is the case in other places), where in fact the `it` must be included within the definied `additionalTests` function body.

Reply:
@nickevansuk my understanding of the convention is:

Any function that begins with `it` spawns `it()` blocks. Therefore, this usage is correct, as `itAdditionalTests` is a function which is expected to spawn `it()` blocks

Reply:
Ah that's not the case at least in these examples...

https://github.com/openactive/openactive-test-suite/blob/95efec494df99ab237efe0aaa2166f4ff055a70b/packages/openactive-integration-tests/test/shared-behaviours/errors.js#L14-L21

https://github.com/openactive/openactive-test-suite/blob/348adcb3d55db809276cac6885b7ae07625aab19/packages/openactive-integration-tests/test/shared-behaviours/common.js#L40-L58

https://github.com/openactive/openactive-test-suite/blob/348adcb3d55db809276cac6885b7ae07625aab19/packages/openactive-integration-tests/test/shared-behaviours/common.js#L24-L27

Reply:
Those functions' names begin with `it` and they spawn `it()` blocks - like the `itAdditionalTests` functions.

I don't think I understand

Reply:
Or perhaps more specifically, the `it*` function is not expected to contain a parameter that requires further `it` blocks... it is itself an `it` block, as the `describe*` functions are describe blocks...